test/QuoterV3.integ.ts

```diff
@@ -0,0 +1,153 @@
+import { expect } from 'chai'
+import { BigNumber } from 'ethers'
+import { QuoterV3 } from '../typechain'
+
+import { JsonRpcSigner } from '@ethersproject/providers'
+import hre, { ethers } from 'hardhat'
+import { encodePath } from './shared/path'
+import { expandTo18Decimals, expandToNDecimals } from './shared/expandTo18Decimals'
+import { FeeAmount, V2_FEE } from './shared/constants'
+
+const V3_FACTORY = '0x1F98431c8aD98523631AE4a59f267346ea31F984'
+const V2_FACTORY = '0x5C69bEe701ef814a2B6a3EDD4B1652CB9cc5aA6f'
+
+const USDC = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48'
+const USDT = '0xdAC17F958D2ee523a2206206994597C13D831ec7'
+const WETH = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'
+const UNI = '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984'
+const DAI = '0x6B175474E89094C44Da98b954EedeAC495271d0F'
+
+/// @dev basic V2 routes
+const DAI_V2_UNI_V2_WETH = encodePath([DAI, UNI, WETH], [V2_FEE, V2_FEE])
+const USDC_V2_UNI_V2_WETH = encodePath([USDC, UNI, WETH], [V2_FEE, V2_FEE])
+
+/// @dev basic V3 routes
+const USDC_V3_500_USDT = encodePath([USDC, USDT], [FeeAmount.LOW])
+const UNI_V3_3000_WETH = encodePath([UNI, WETH], [FeeAmount.MEDIUM])
+
+/// @dev stablecoin IL routes
+const USDT_V3_500_DAI_V2_USDC = encodePath([USDT, DAI, USDC], [FeeAmount.LOW, V2_FEE])
+const DAI_V3_100_USDC_V2_USDT = encodePath([DAI, USDC, USDT], [100, V2_FEE])
+
+/// @dev erc20 IL routes
+// V3 - V2
+const UNI_V3_3000_WETH_V2_DAI = encodePath([UNI, WETH, DAI], [FeeAmount.MEDIUM, V2_FEE])
+const USDC_V3_3000_UNI_V2_WETH = encodePath([USDC, UNI, WETH], [FeeAmount.MEDIUM, V2_FEE])
+// V2 - V3
+const UNI_V2_WETH_V3_3000_DAI = encodePath([UNI, WETH, DAI], [V2_FEE, FeeAmount.MEDIUM])
+
+/// @dev complex IL routes
+// (use two V3 pools)
+const DAI_V3_3000_UNI_V2_USDT_V3_3000_WETH = encodePath(
+  [DAI, UNI, USDT, WETH],
+  [FeeAmount.MEDIUM, V2_FEE, FeeAmount.MEDIUM]
+)
+// (use two V2 pools)
+const DAI_V3_3000_UNI_V2_USDT_V2_WETH = encodePath([DAI, UNI, USDT, WETH], [FeeAmount.MEDIUM, V2_FEE, V2_FEE])
+
+describe('QuoterV3 integration tests', () => {
+  let quoterV3: QuoterV3
+  let alice: JsonRpcSigner
+
+  before(async () => {
+    const QuoterV3Factory = await ethers.getContractFactory('QuoterV3')
+    quoterV3 = (await QuoterV3Factory.deploy(V3_FACTORY, V2_FACTORY, WETH)) as QuoterV3
+    console.log('quoterV3.address: ', quoterV3.address)
+  })
+
+  /**
+   * Values only valid at block 14390000, we should not be running a local node of hardhat to test against, but rather using
+   * the jest-environment-hardhat plugin. TODO
```

Comment:
is this still todo? Can we have a test here that checks the block number

Reply:
Ah yeah it is still TODO. I can use the plugin we use in SOR now

Reply:
Nvm, figured out how to do it without the plugin